apps/backend/src/utils/nginx_parser.py

```
@@ -0,0 +1,553 @@
+    def parse_config_file(self, file_path: str) -> Dict[str, Any]:
+        """
+        Parse a complete nginx configuration file
+        
+        Args:
+            file_path: Path to the nginx config file
+            
+        Returns:
+            Dict containing parsed configuration data
+        """
+        try:
+            with open(file_path, 'r', encoding='utf-8') as f:
```

Ruff (0.12.2)

Unnecessary mode argument

Remove mode argument

(UP015)